prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -0,0 +1,80 @@
+"""
+PYTHONPATH=. streamlit run prediction_market_agent/agents/microchain_agent/app.py
+
+Tip: if you specify PYTHONPATH=., streamlit will watch for the changes in all files, isntead of just this one.
+"""
+import streamlit as st
+
+from prediction_market_agent.utils import APIKeys, streamlit_asyncio_event_loop_hack
+
+# Imports using asyncio (in this case mech_client) cause issues with Streamlit
+streamlit_asyncio_event_loop_hack()
+
+from microchain import Agent
+from prediction_market_agent_tooling.markets.markets import MarketType
+from prediction_market_agent_tooling.tools.costs import openai_costs
+
+from prediction_market_agent.agents.microchain_agent.microchain_agent import get_agent
+
+
+def check_api_keys() -> None:
+    keys = APIKeys()
+    if not keys.OPENAI_API_KEY:
+        st.error("No OpenAI API Key provided via env var/secret.")
+        st.stop()
+    elif not keys.BET_FROM_PRIVATE_KEY:
+        st.error("No wallet private key provided via env var/secret.")
+        st.stop()
+
+
+def run_agent(agent: Agent, iterations: int) -> None:
+    with openai_costs(model) as costs:
```

Comment:
Should this be taken from global variable?

Reply:
Ah good spot. I've added a model arg to this function